File: BackEnd/handler.js
//import { connection } from "./connection";
//import { mysql, connect } from './connection.js'
const mysql = require('mysql')

//Creates a connection to my local the AWS mySQL database.
/*
var connection = mysql.createConnection({
    host: "localhost",
    user: "testuser",
    password: "SecuritE-mail",
    port: "3306",
    database: "securite-mail"

});
*/
var phrases = [];

function connect() {

    var connection = mysql.createConnection({
        host: "localhost",
        user: "testuser",
        password: "SecuritE-mail",
        port: "3306",
        database: "securite-mail"
    });


    let phrases = [];
    connection.connect(function(err) {
        if(err) {
         console.error('Database connection failed: ' + err.stack);
            return;
     }
        console.log('Connected to database.')
        connection.query("SELECT phrase FROM phrases", function (err, result, fields) {
         if (err) throw err;
            for(i = 0; i < result.length; i++) {
             phrases[i] = result[i].phrase;
            //console.log("index " + i + " " + phrases[i])
             }
         });
        connection.end(function(err) {
            if (err) {
                return console.log('error:' + err.message);
            }
            console.log('Close the database connection.');
        });  
    });

    return phrases;
}
  
phrases = connect();


/**
 * Converts the email into a readable form for
 * the application to be able to use
 * @param {Email object} email 
 */
function interpretEmail(email) {
    //this will change the email into a readable format to process
    let strEmail = email.toUpperCase();
    return strEmail;
}

/**
 * Takes the email and changes it from plaintext
 * to a hashed version found inside the database
 * @param {Email Object} email 
 */
function toHash(email) {
    //hash given string
    let plaintxt = interpretEmail(email);
    let hashText = plaintxt;
    return hashText
}

/**
 * Given an email this will scan through and return the appropriate score
 * 
 * @param {Email Object} email 
 * @param {Array} phrases 
 */
function calcuateScore(email, phrases) {
    //calculate the score of all given words

    //txt is the hashed version of the readable email
    //score is score, duh
    let txt = toHash(interpretEmail(email));
    let score = 0;

    //for each phrase inside the array
    //find an occurence and update the score
    for(i = 0; i < phrases.length; i++) {
        let pos = txt.indexOf(phrases[i].toUpperCase());
        if(pos >= 0) {
            console.log(phrases[i])
            score++;
        }
    }

    return score;
}

function rating(score) {
    if (score == 0) {
        console.log("Risk Level: None");
    } else if (score == 1) {
        console.log("Risk Level: Low");
    } else if (score == 2) {
        console.log("Risk Level: Medium");
    } else {
        console.log("Risk Level: High")
    }
    return score;
}
/*
var phrases = ["Track capacity","Intercept range","Radar range",
    "Missile type","Missile flight","Missile capability","Radar capability",
    "Missile inventory","Ship Capability","Missile Range","Missile Capacity","Track Types", "Trackable Object"]
*/

var emailtst = "Good afternoon Professor Chu, We are pleased to tell you about the project involving missile flight."
        + " There is a missile there and a missile here. The Radar range seems to be be off but that is nothing we cannot handle." 
        + " Therefore, we will no longer meander around and get to work. Ship Capability. Regards, Elon Musk";


        setTimeout(() => {console.log(rating(calcuateScore(emailtst, phrases)));}, 3000);

File: BackEnd/handler.py
import os
import hashlib
import re
import pymysql.cursors


# Connection is the variable that connects to the database
# It uses local host because it is hosted on the same
# server as the database
# User is the username and password is the associated password
# database is the name of the database being accessed
connection = pymysql.connect(host='localhost', user='SYSTEM_USER',
                                       password='SecuritE-mail', database='phrases', charset="latin1",
                                       cursorclass=pymysql.cursors.DictCursor, port=3306)
phrases = []
# This try block is the execution of the connection
# It will launch the query and store the results
# as a list of dictionaries inside result.
# The key for each dictionary is the row queried for
# EX: SELECT row FROM database.table
try:
	cursor =  connection.cursor()
	cursor.execute("SELECT phrases FROM phrases.phrases")
	result = cursor.fetchall()
	for i in range(0, len(result)):
		phrases.append(result[i]['phrases'])
except:
	print("failure")



# This is the hashing function
# It was a stretch goal and there was a basic version experimented with
# However, if in the future a hashing function were to be implemented
# There is a ready frame for it here
def toHash(email):
    """
    This is the hashing function that will change
    the string from plaintext to an encrypted string

    :param email: the string version of the email
    :return: hashed version of the email
    """
    hashed = email
    #email = re.findall(r"[\w']+", email)  # Ignore whitespace and punctuation
    #for i in email:  # Loop through individual in provided text
    #    salt = os.urandom(32)  # Generate a random salt
    #    i = hashlib.pbkdf2_hmac(
    #        'sha256',  # Hash algorithm to use
    #        i.encode('utf-8'),  # Convert pasword to bytes
    #        salt,  # Provide the salt
    #        100000)  # Number of iterations
    #    hashed += i.hex()  # Appends hashed version of word as hex characters
    return hashed


def calculateScore(email: str, phrases: list):
    """
    Calculates the score for the email

    :param email: hashed version of email
    :param phrases: list of all the words
    :return: the score (pings) of the email
    """
	# txt is the email but put into upper case
	# At the moment toHash does no hashing
    txt = toHash(email).upper()
    score = 0

	# Finds occurences of each word in the retrieved database
	# then finds an instance of it inside the email string
	# does not count repeats
    for i in range(0, len(phrases)):
        pos = txt.find(phrases[i].upper())
        if pos > -1:
            score = score + 1

    return score


# Everything is testing below this
# ======================================================================================================================
def rating(score):
    if score == 0:
        print("Risk Level: None")
    elif score == 1:
        print("Risk Level: Low")
    elif score == 2:
        print("Risk Level: Medium")
    else:
        print("Risk Level: High")

    return score


# ======================================================================================================================

hashedPhrases = []

tst = input()
print(calculateScore(tst, phrases))
